csharp/src/Drivers/Apache/Hive2/HiveServer2Connection.cs

```diff
@@ -27,21 +26,21 @@
 
 namespace Apache.Arrow.Adbc.Drivers.Apache.Hive2
 {
-    public abstract class HiveServer2Connection : AdbcConnection
+    internal abstract class HiveServer2Connection : AdbcConnection
```

Comment:
@CurtHagenlocher 

The motivation behind this change is ...

1. These Connection/Statement classes should not be creatable/visible outside of the library. The entry point is/should be the Driver. We should go back and change BigQuery - in my opinion.
2. In turn, this makes it "easier" to expose those methods that need to be used outside the class. I was finding that I had to method-by-method find the ones that needed to be set to `internal`.
3. This was also a best-practice for JDBC driver development I've worked on.

Of course, this is just a suggestion. If you feel it would make the PR easier to review, I can revert these changes.

Reply:
If for some reason there were driver-specific functionality that we wanted to expose then the class would need to be public. That said, nothing prevents this from being changed later, and defaulting to publishing only the types we need to is probably best.

Reply:
Reverted change from `public` back to `internal`.